domains/Discussions/Tests/Feature/QuestionsIndexTest.php

```diff
@@ -0,0 +1,218 @@
+        QuestionFactory::new([
+            'title' => 'laravel-Pt',
+        ])
+            ->create();
+
+        $this->json('GET', route('discussions.questions.index', [
+            'title' => 'LArAvEL-pT',
+        ]))
+            ->seeJsonContains([
+                'to' => 2,
+            ]);
+    }
+
+    /** @test */
+    public function it_search_by_created_date(): void
+    {
+        QuestionFactory::new([
+            'created_at' => Carbon::now()->subYears(2),
+        ])
+            ->create();
+        QuestionFactory::new([
+            'created_at' => Carbon::now()->subYears(3),
+        ])
+            ->create();
+
+        $this->json('GET', route('discussions.questions.index', [
+            'created[from]' => Carbon::now()->subMonth()->subYears(2)->toDateString(),
+            'created[to]' => Carbon::now()->addMonth()->subYears(2)->toDateString()
+        ]))
+            ->seeJsonContains([
+                'to' => 1,
+            ]);
+
+        $this->json('GET', route('discussions.questions.index', [
+            'created[from]' => Carbon::now()->subMonth()->subYears(3)->toDateString(),
+            'created[to]' => Carbon::now()->addMonth()->subYears(2)->toDateString()
+        ]))
+            ->seeJsonContains([
+                'to' => 2,
+            ]);
+    }
+
+    /** @test */
+    public function it_search_by_resolved(): void
+    {
+        QuestionFactory::new([
+            'resolved_at' => Carbon::now(),
+        ])
+            ->count(5)
+            ->create();
+
+        $this->json('GET', route('discussions.questions.index'))
+            ->seeJsonContains([
+                'to' => 15,
+            ]);
+
+        $this->json('GET', route('discussions.questions.index', [
+            'resolved' => true,
+        ]))
+            ->seeJsonContains([
+                'to' => 5,
+            ]);
+
+        $this->json('GET', route('discussions.questions.index', [
+            'resolved' => false,
+        ]))
+            ->seeJsonContains([
+                'to' => 10,
+            ]);
+    }
+
+    /** @test */
+    public function it_fails_by_validations(): void
+    {
+        $this->get(route('discussions.questions.index', [
+            'author' => 'author',
+        ]))
+            ->assertResponseStatus(Response::HTTP_UNPROCESSABLE_ENTITY);
+
+        $this->get(route('discussions.questions.index', ['resolved' => 12332]))
+            ->assertResponseStatus(Response::HTTP_UNPROCESSABLE_ENTITY);
+
+        $this->get(route('discussions.questions.index', [
+            'created[from]' => Carbon::now()->toDateString(),
+        ]))
+            ->assertResponseStatus(Response::HTTP_UNPROCESSABLE_ENTITY);
+
+        $this->get(route('discussions.questions.index', [
+            'created[from]' => Carbon::now()->toDateString(),
+            'created[to]' => Carbon::now()->subDay()->toDateString(),
+        ]))
+            ->assertResponseStatus(Response::HTTP_UNPROCESSABLE_ENTITY);
```

Comment:
We can take advantage of PHPUnit's Data Providers to test all possible combinations of searchable properties, and it'll also future-proof this a little for the future.

Reply:
ref: https://phpunit.readthedocs.io/en/9.3/writing-tests-for-phpunit.html?highlight=data%20provider#data-providers